### PR TITLE
samba.setup: Indicate the use of VFS module in share name

### PR DESCRIPTION
**Pull request by anoopcs9**

### CODE REVIEW

**Review by xhernandez — COMMENTED** [review]
I like the idea, but I think it would be better to make it more generic.

For example by setting the `share_name` to something generic, like `share`, or `replicate`/`disperse` in the case of gluster, and then always append the backend and the variant to that name, like:

    share-backend-variant

File: vagrant/ansible/roles/samba.setup/tasks/main.yml [hunk 1]
```diff
@@ -36,16 +36,18 @@
...
       with_items: "{{ samba_shares }}"
 
     - name: Include share configuration into smb.conf
       lineinfile:
         path: "/etc/samba/smb.conf"
-        line: "include = /etc/samba/smb.shares/{{ item.share_name }}.conf"
+        line: "include = /etc/samba/smb.shares/{{ name }}.conf"
         insertafter: EOF
+      vars:
+        name: "{{ item.share_name }}-{{ config.be.name }}-{{ config.be.variant }}"
```
Comment (xhernandez) [inline vagrant/ansible/roles/samba.setup/tasks/main.yml:50]: Maybe it could be cleaner to set this var at the _block_ level and use it in all subtasks.

Comment (anoopcs9) [reply vagrant/ansible/roles/samba.setup/tasks/main.yml]: Done.

**Review by xhernandez — APPROVED** [review]

**Review by phlogistonjohn — APPROVED** [review]
Looks OK to me

**Comment by anoopcs9** [comment]
> /retest centos-ci/glusterfs

Failure with GlusterFS installation should be fixed with #46.

**Comment by anoopcs9** [comment]
Sample output:

> "testcases/consistency/test_consistency.py::test_consistency[192.168.123.11-replicate-glusterfs-default] PASSED [ 25%]",
"testcases/consistency/test_consistency.py::test_consistency[192.168.123.10-disperse-glusterfs-default] PASSED [ 50%]",
"testcases/consistency/test_consistency.py::test_consistency[192.168.123.11-disperse-glusterfs-default] PASSED [ 75%]",
"testcases/consistency/test_consistency.py::test_consistency[192.168.123.10-replicate-glusterfs-default] PASSED [100%]",

> "testcases/consistency/test_consistency.py::test_consistency[192.168.123.10-share-xfs-default] PASSED [100%]",

> "testcases/consistency/test_consistency.py::test_consistency[192.168.123.11-share-cephfs-vfs] PASSED [ 33%]",
"testcases/consistency/test_consistency.py::test_consistency[192.168.123.10-share-cephfs-vfs] PASSED [ 66%]",
"testcases/consistency/test_consistency.py::test_consistency[192.168.123.12-share-cephfs-vfs] PASSED [100%]",

**Pull request merged**